apps/charterafrica/src/lib/data/common/index.js

```diff
@@ -171,48 +171,47 @@ export async function processPageFellowships({ blocks }) {
   });
 }
 
-export async function processPageNews({ blocks }) {
-  // TODO(kilemensi): Pull data from CMS
-  blocks.push({
-    slug: "featured-post",
-    category: "News",
-    title: "News Story title goes here and spans over second line",
-    excerpt:
-      "Lorem ipsum dolor sit amet consectetur adipiscing elit tempus nibh cursus, urna porta sagittis non eget taciti nunc sed felis dui, praesent ullamcorper facilisi euismod ut in platea laoreet integer. Lorem ipsum dolor sit amet consectetur",
-    date: "2020-10-10 10:10:10",
-    author: "Author Name",
-    image: {
-      url: "/images/featured_post.jpg",
-      alt: "Featured Post",
-    },
+export async function processPageNews(page, api) {
+  const { blocks, breadcrumbs = [] } = page;
+  const { docs } = await api.getCollection("news", {
+    where: { _status: { equals: "published" } },
+  });
+
+  const processArticle = (data) => ({
+    ...data,
+    author: data?.authors?.map(({ fullName }) => fullName).join(", ") ?? null,
+    image: data?.coverImage ?? null,
+    date: new Date(data?.publishedOn).toUTCString(),
     link: {
-      href: "/knowledge/news",
+      href: `${breadcrumbs[breadcrumbs.length - 1]?.url}/${data?.id}`,
```

Comment:
1. To avoid having bogus/undefined `href`, I think lets check if we indeed do have breadcrumbs. If not, `href` should be `null` so that in the component, we don't use `Link`.
2. I think `slugs` make better URLs than `ids`.